lib/src/switch/Switch.tsx

```diff
@@ -27,25 +25,49 @@ const DxcSwitch = ({
   const [switchId] = useState(`switch-${uuidv4()}`);
   const labelId = `label-${switchId}`;
   const [innerChecked, setInnerChecked] = useState(defaultChecked ?? false);
+  const [hasLabel] = useState((label !== "" && label !== null && label !== undefined) ?? false);
+
   const colorsTheme = useTheme();
   const translatedLabels = useTranslatedLabels();
   const backgroundType = useContext(BackgroundColorContext);
+  const refTrack = useRef(null);
+
+  const handleOnKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    switch (event.keyCode) {
+      case 13: // enter
+        event.preventDefault();
+        refTrack.current.focus();
+        const isChecked = !(checked ?? innerChecked);
+        setInnerChecked(isChecked);
+        if (typeof onChange === "function") {
+          onChange(isChecked);
+        }
+        break;
+    }
+  };
 
   const handlerSwitchChange = (event) => {
     if (checked === undefined) {
       const isChecked = event.target.checked ?? !innerChecked;
       setInnerChecked(isChecked);
-      onChange?.(isChecked);
-    } else onChange?.(!checked);
+      if (typeof onChange === "function") {
+        onChange(isChecked);
+      }
+    } else {
+      if (typeof onChange === "function") {
+        onChange(!checked);
+      }
+    }
   };
 
   const labelComponent = (
```

Comment:
labelComponent is being rerendered each time the component as a whole rerenders, due to being stored as a variable and not being in the jsx returned code or outside the component. 

I think the best solutions is add it to the JSX tree.